doc/man/man5/strsctp.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH @PACKAGE_UCNAME@ 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System File Formats"
.SH NAME
.B @PACKAGE@
\- the @PACKAGE_TITLE@ package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@ - @VERSION@ )
package for
.BR "Linux Fast-STREAMS" .
.[
lfs
.]
.PP
OpenSS7
.I STREAMS
add-on packages are separate add-on packages to the
.B Linux Fast-STREAMS
.[
lfs
.]
release that provide additional functionality and protocol support not traditionally found on UNIX
systems, or which have traditionally fallen into the domain of value-add system software, or, in
this case, where the functionality was optional or differently handled in a number of implementations.  The
.B @PACKAGE_TITLE@
package is one of several optional add-on packages that provide enhanced
.I STREAMS
support beyond that provided by the base
.I STREAMS
package.  This package provides support for Stream Control Transmision Protocol based on the
.IR "Transport Provider Interface (TPI)" ,
.BR tpi (7),
not provided by the base
.I STREAMS
package.
.\"
.\"
.SH DESCRIPTION
.PP
The
.B @PACKAGE_TITLE@
package provides
.IR STREAMS (9)
modules and drivers, libraries, programs, initialization scripts, and daemons that
implement and provide
the Stream Control Transmission Protocol (SCTP).
.[
rfc2960
.]
.[
rfc4460
.]
.PP
The package currently provides the following
.IR STREAMS (9)
modules and drivers:
.PP
.TS
tab(:);
cw(1u)1lbcliss.
\-:streams_sctp.ko:driver:/dev/sctp_n, /dev/sctp_t
\-:streams_tpiperf.ko:module:\(lqtpiperf\(rq
.TE
.TP
.BR sctp (4)
The
.BR sctp (4)
driver is a
.IR STREAMS (9)
pseudo-device driver that provides
.BR /dev/sctp_n " and " /dev/sctp_t
devices.  See the
.BR sctp (4)
manual page for more information.
.TP
.BR tpiperf (4)
The
.BR tpiperf (4)
module is a pushable
.IR STREAMS (9)
module that provides special capabilities for performance testing of TPI
providers.
See the
.BR tpiperf (4)
manual page for more information.
.PP
The package builds and installs the following shared object libraries:
.TP
.BR libsctp (3)
The
.BR libsctp (3)
library provides compatibility with the suggested SCTP sockets API.
This library requires the
.BR strsock (5)
package.  It provides the following functions:
.RS
.PP
.TS
tab(:);
lb1lw(1u)1lsss.
sctp_bindx\fR(3):\-:T{
bind SCTP socket to multiple addresses.
T}
sctp_connectx\fR(3):\-:T{
connect SCTP socket to multiple addresses.
T}
getsockopt\fR(3):\-:T{
set SCTP socket options.
T}
setsockopt\fR(3):\-:T{
get SCTP socket options.
T}
.TE
.RE
.PP
This library is incomplete.
.PP
The
.B @PACKAGE_TITLE@
package includes kernel modules,
.I SVR 4.2
.[
svr42
.]
.I STREAMS
drivers, modules, libraries, utilities, test programs, daemons, and
development environment for the development and execution of
.B @PACKAGE_TITLE@
protocol stack components and applications for the
.B Linux Fast-STREAMS
environment.
.PP
This distribution is only currently applicable to
.B Linux
2.4 and 2.6 kernels and was targeted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
architectures, but should build and install for other architectures as well.
.PP
This package is complete and has been in production use for
.I Linux Fast-STREAMS
for a number of years.
.\"
.\"
.SH "PACKAGE CONTENTS"
.PP
The
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@ - @VERSION@ )
package contains the following manual pages:
.PP
.TS
tab(:);
lb1lw(1u)1lssss.
sctp\fR(4):\-:T{
Stream Control Transmission Protocol (SCTP)
T}
sctp_ioctl\fR(4):\-:T{
specialized SCTP input-output controls.
T}
sctp_n\fR(4):\-:T{
the
.BR sctp (4)
Network Provider Interface (NPI),
.[
npi revision 2.0.0
.]
.BR npi (7)
Connection Oriented Network Service device.
T}
sctp_n\fR(4):\-:T{
the
.BR sctp (4)
Network Provider Interface (NPI),
.[
npi revision 2.0.0
.]
.BR npi (7)
Connection Oriented Network Service device.
T}
sctp_t\fR(4):\-:T{
Transport Provider Interface (TPI),
.[
tpi revision 2.0.0
.]
.BR tpi (7)
Connection Oriented Transport Service device with Orderly Release.
T}
tpiperf\fR(4):\-:T{
Transport Provider Interface (TPI),
.[
tpi revision 2.0.0
.]
.BR tpi (7)
performance testing
.IR STREAMS (9)
module.
T}
strsctp\fR(5):\-:T{
the
.B @PACKAGE@
package.
T}
npi_sctp\fR(7):\-:T{
Network Provider Interface (NPI),
.[
npi revision 2.0.0
.]
.BR npi (7),
.IR "Corrigendum for SCTP Conformance" .
T}
tpi_sctp\fR(7):\-:T{
Transport Provider Interface (TPI),
.[
tpi revision 2.0.0
.]
.BR tpi (7),
.IR "Corrigendum for SCTP Conformance" .
T}
xti_sctp\fR(3):\-:T{
X/Open Transport Interface
.[
xti
.]
.BR xti (3),
.IR "Use of the XTI with SCTP" .
T}
strsctp_mknod\fR(8):\-:T{
create or remove SCTP device nodes.
T}
test-etsi_n\fR(8):\-:T{
SCTP NPI interface ETSI validate test case executable.
T}
test-etsi_t\fR(8):\-:T{
SCTP TPI interface ETSI validate test case executable.
T}
test-interop\fR(8):\-:T{
10th SCTP Interop test program.
T}
test-sctp\fR(8):\-:T{
SCTP quick check test.
T}
test-sctp_n2\fR(8):\-:T{
SCTP NPI interface validation test case executable
T}
test-sctp_n\fR(8):\-:T{
SCTP NPI interface validation test case executable
T}
test-sctp_nc\fR(8):\-:T{
SCTP NPI Client test program
T}
test-sctp_ns\fR(8):\-:T{
SCTP NPI Server test program
T}
test-sctp_t\fR(8):\-:T{
SCTP TPI interface validation test case executable.
T}
.TE
.\"
.\"
.\".SH NOTICES
.\".PP
.\"
.\"
.\".SH EXAMPLES
.\".PP
.\"
.\"
.SH MODULES
.PP
The
.B @PACKAGE@
package provides the following
.IR STREAMS (9)
moduels:
.HP
.BR tpiperf (4)\ \-\ \c
TPI performance test module.
.\"
.\"
.SH DEVICES
.PP
The
.B @PACKAGE@
package provides the following
.IR STREAMS (9)
devices:
.HP
.BR sctp (4)\ \-\ \c
The SCTP pseudo-device driver.
.\"
.\"
.SH FILES
.PP
.TP
.BI /lib/modules/ `uname\ -r` /streams/streams_sctp.ko
The
.BR sctp (4)
.IR STREAMS (9)
driver kernel module.
.TP
.BI /lib/modules/ `uname\ -r` /streams/streams_tpiperf.ko
The
.BR tpiperf (4)
.IR STREAMS (9)
module kernel module.
.\"
.\"
.SH PREREQUISITES
.PP
The
.B @PACKAGE@
package requires the following packages to be installed:
.TP
.BR streams (5)
The
.B Linux Fast-STREAMS
base package.
.TP
.BR strcompat (5)
The
.B Linux Fast-STREAMS
compatibility modules package.
.TP
.BR strxns (5)
The
.B Linux Fast-STREAMS
X/Open Networking Services package.
.TP
.BR strxnet (5)
The
.B Linux Fast-STREAMS
XTI/TLI Library package.
.TP
.BR strnsl (5)
The
.B Linux Fast-STREAMS
Network Selection Library package.
.TP
.BR strsock (5)
The
.B Linux Fast-STREAMS
Sockets Library package.
.\"
.\"
.SH NOTICES
.PP
.I Production
releases are releases that have received exhaustive testing within the project
and validated on specific distributions and architectures.
.I Production
release packages do not ship with known defects.  Please rememaber that there
is
.B NO WARRANTY
on public release packages.
.PP
.I Production
packages ship containing a list of validated distributions and architectures.
Full regression testing of any maintenance changes is performed.  Suitability
of software in this category for production use on the specified target
distributions and architectures is at the discretion of the user.  It should
not be necessary to perform validation tests on the set of supported target
systems before considering production use.
.PP
.B @PACKAGE_TITLE@
could possibly contain unknown defects.  This is a
.I production
release.  Nevertheless, some remaining unknown defects could possibly be
harmful.  Validation testing has been performed by the
.I OpenSS7 Project
and external entities on this software for the set of systems listed in the
release notes.  Nevertheless, the software might still fail to configure or
compile on other systems.  The
.I OpenSS7 Project
recommends that you validate this software for your target system before using
this software.  Use at your own risk.  Remember that there is
.BR "NO WARRANTY" .
(See the section
.RI \(lq "NO WARRANTY" \(rq
under the
.RI \(lq "GNU General Public License" \(rq,
in the manual.)
.PP
This software is
.I production
software.  As such, it is stable on validated systems but might still crash
your kernel in unique circumstances.  Installation of the software on a
non-validated distribution might mangle your header files or Linux
distribution in such a way as to make it unusable and rebooting the system
might not repair the problem.  You possibly lose all the data on your system.
Because this software stands a chance of crashing your kernel, the resulting
unstable system could possibly destroy computer hardware or peripherals making
them unusable.  You might void the warranty on any system on which you run
this software.
.B YOU HAVE BEEN WARNED.
.\"
.\"
.SH BUGS
.PP
The
.B @PACKAGE@
package has been tested with a wide array of production
.B Linux
distributions, kernels, and target architectures, and is considered production grade by
.IR "The OpenSS7 Project" .
At the time of release, the
.B @PACKAGE@
package had no known bugs.  Problems can be reported to
.RB < bugs@openss7.org >.
Please, however, use the included
.B send-pr
script for generating actionable problem reports.
Please see the file
.RI \(lq PROBLEMS \(rq
in the release for problem reporting guidelines.
.\"
.\"
.SH "SEE ALSO"
.PP
.BR sctp (4),
.BR strsctp (5),
.BR strsctp_mknod (8),
.BR STREAMS (9).
.\"
.\"
.SH COMPATIBILITY
.PP
.B @PACKAGE_TITLE@
is compatible with other SCTP implementations and was interoperability tested
at the IETF supported 2nd and 9th and 10th SCTP interoperatbility events,
twice at ETSI at the headquarters in Sophia Antipolis, France, and once at the
University of British Columbia, Canada.
.\"
.\"
.SH CONFORMANCE
.PP
RFC 2960,
.[
rfc2960
.]
RFC 3309,
.[
rfc3309
.]
RFC 4460,
.[
rfc4460
.]
NPI,
.[
npi revision 2.0.0
.]
TPI,
.[
tpi revision 2.0.0
.]
XTI,
.[
xti
.]
XNS 5.2.
.[
xns
.]
To date\(emasside from the protocol specification\(emthere is no standard for the
interface or implementation of SCTP.
Conformance is tested using the
.BR test-sctp (8),
.BR test-etsi_n (8),
.BR test-etsi_t (8),
.BR test-interop (8),
.BR test-sctp_n2 (8),
.BR test-sctp_n "(8) and "
.BR test-sctp_t (8),
test case executables and the
.I @PACKAGE_TITLE@
.B autotest
test suite.
.\"
.\"
.SH HISTORY
.PP
The
.B @PACKAGE_TITLE@
package
is one of the original kernel implementations of SCTP.
.PP
The
.IR STREAMS (9)
version of SCTP was separated from the
.BR strss7 (5)
package to become the
.BR strsctp (5)
package in the 0.8.2 releases.
.PP
The
.B @PACKAGE_TITLE@
package is a stable, production grade release.
.\"
.\"
.SH "FUTURE DIRECTIONS"
.PP
The SCTP package is a stable production release and is largely in
maintenance mode.  There are, nevertheless, several items remaining to be
done.  See the
.RI \(lq TODO \(rq
file in the release for detailed information on further development.
.\"
.\"
.[
$LIST$
.]
.\"
.TI


NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/util/init/strsctp.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strsctp:	start and stop strsctp subsystem
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strsctp
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		character devices are present in the /dev directory and \
#		that the STREAMS SCTP subsystem is configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strsctp
# Required-Start: streams $network
# Required-Stop: streams $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS SCTP subsystem
# License: GPL
# Description:	This STREAMS SCTP init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS SCTP character
#	devices are present in the /dev directory and that the STREAMS SCTP
#	subsystem is configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strsctp'
config="/etc/default/$name"
desc="the STREAMS SCTP subsystem"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRSCTP_MKNOD" ; then
    for STRSCTP_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $STRSCTP_MKNOD ] ; then
	    break
	else
	    STRSCTP_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$STRSCTP_PRELOAD"       ] || STRSCTP_PRELOAD=""
[ -n "$STRSCTP_DRIVERS"       ] || STRSCTP_DRIVERS="streams-sctp"
[ -n "$STRSCTP_MODULES"       ] || STRSCTP_MODULES="streams-tpiperf"
[ -n "$STRSCTP_MAKEDEVICES"   ] || STRSCTP_MAKEDEVICES="yes"
[ -n "$STRSCTP_REMOVEDEVICES" ] || STRSCTP_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRSCTP_MKNOD" ] && STRSCTP_MAKEDEVICES="no"
[ -z "$STRSCTP_MKNOD" ] && STRSCTP_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRSCTP_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRSCTP_MKNOD" -a ":$STRSCTP_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS SCTP devices: "
	$STRSCTP_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRSCTP_MKNOD" -a ":$STRSCTP_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS devices: "
	$STRSCTP_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRSCTP_PRELOAD $STRSCTP_DRIVERS $STRSCTP_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.17  2008-10-21 22:42:18  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.16  2008-04-28 23:13:30  brian
# - updated headers for release
#
# Revision 0.9.2.15  2007/08/14 06:23:04  brian
# - GPLv3 header update
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
